src/main/java/com/mparticle/kits/IterableKit.java

```diff
@@ -2,26 +2,69 @@
-
+import android.util.Log;
+
+import com.google.firebase.messaging.RemoteMessage;
+import com.iterable.iterableapi.IterableApi;
+import com.iterable.iterableapi.IterableConfig;
+import com.iterable.iterableapi.IterableConfigHelper;
+import com.iterable.iterableapi.IterableConstants;
+import com.iterable.iterableapi.IterableFirebaseMessagingService;
+import com.iterable.iterableapi.IterableHelper;
 import com.mparticle.AttributionResult;
 import com.mparticle.MParticle;
+import com.mparticle.identity.MParticleUser;
+import com.mparticle.kits.iterable.BuildConfig;
 
+import java.lang.ref.WeakReference;
+import java.util.HashMap;
 import java.util.HashSet;
 import java.util.List;
 import java.util.Map;
 import java.util.Set;
 
 
-public class IterableKit extends KitIntegration implements KitIntegration.ActivityListener, KitIntegration.ApplicationStateListener {
+public class IterableKit extends KitIntegration implements KitIntegration.ActivityListener, KitIntegration.ApplicationStateListener, KitIntegration.IdentityListener, KitIntegration.PushListener {
+    private Set<String> previousLinks = new HashSet<String>();
+    private boolean mpidEnabled = false;
+
+    private static IterableConfig customConfig;
 
+    private final static String SETTING_API_KEY = "apiKey";
+    private final static String SETTING_GCM_INTEGRATION_NAME = "gcmIntegrationName";
+    private final static String SETTING_USER_ID_FIELD = "userIdField";
 
-    private String attributionUrl;
-    private Set<String> previousLinks = new HashSet<String>();
+    private final static String IDENTITY_CUSTOMER_ID = "customerId";
+    private final static String IDENTITY_MPID = "mpid";
+
+    private final static String INTEGRATION_ATTRIBUTE_KIT_VERSION_CODE = "Iterable.kitVersionCode";
+    private final static String INTEGRATION_ATTRIBUTE_SDK_VERSION = "Iterable.sdkVersion";
+
+    /**
+     * Set a custom config to be used when initializing Iterable SDK
+     * @param config `IterableConfig` instance with configuration data for Iterable SDK
+     */
+    public static void setCustomConfig(IterableConfig config) {
+        customConfig = config;
+    }
 
     @Override
     protected List<ReportingMessage> onKitCreate(Map<String, String> settings, Context context) {
-        checkForAttribution();
+        try {
+            checkForAttribution();
+
+            String userIdField = settings.get(SETTING_USER_ID_FIELD);
+            mpidEnabled = userIdField != null && userIdField.equals(IDENTITY_MPID);
+
+            IterableConfig.Builder configBuilder = IterableConfigHelper.createConfigBuilderFromIterableConfig(customConfig);
+            configBuilder.setPushIntegrationName(settings.get(SETTING_GCM_INTEGRATION_NAME));
+            IterableApi.initialize(context, settings.get(SETTING_API_KEY), configBuilder.build());
+            initIntegrationAttributes();
+        } catch(Exception e) {
+            Log.e("IterableKit", "Iterable Kit initialization exception", e);
```

Comment:
The API contract is to throw an exception here, which we will catch and log the message of.